airbyte-integrations/connectors/source-notion/unit_tests/test_streams.py

```diff
@@ -76,3 +77,61 @@ def test_backoff_time(patch_base_class):
     stream = NotionStream(config=MagicMock())
     expected_backoff_time = None
     assert stream.backoff_time(response_mock) == expected_backoff_time
+
+
+def test_users_request_params(patch_base_class):
+    stream = Users(config=MagicMock())
+
+    # No next_page_token. First pull
+    inputs = {"stream_slice": None, "stream_state": None, "next_page_token": None}
+    expected_params = {"page_size": 100}
+    assert stream.request_params(**inputs) == expected_params
+
+    # When getting pages after the first pull.
+    inputs = {"stream_slice": None, "stream_state": None, "next_page_token": {"next_cursor": "123"}}
+    expected_params = {"start_cursor": "123", "page_size": 100}
+    assert stream.request_params(**inputs) == expected_params
+
+
+def test_user_stream_handles_pagination_correclty(requests_mock):
+    """
+    Test shows that Users stream uses pagination as per Notion API docs.
+    """
+
+    response_body = {
+        "object": "list",
+        "results": [{"id": f"{x}"} for x in range(100)],
+        "next_cursor": "test_cursor_1",
+        "has_more": True,
+        "type": "user"
+    }
+    url = "https://api.notion.com/v1/users?page_size=100"
+    requests_mock.get(url, json=response_body)
+
+    response_body = {
+        "object": "list",
+        "results": [{"id": f"{x}"} for x in range(100, 200)],
+        "next_cursor": "test_cursor_2",
+        "has_more": True,
+        "type": "user"
+    }
+    url = "https://api.notion.com/v1/users?page_size=100&start_cursor=test_cursor_1"
+    requests_mock.get(url, json=response_body)
+
+    response_body = {
+        "object": "list",
+        "results": [{"id": f"{x}"} for x in range(200, 220)],
+        "next_cursor": None,
+        "has_more": False,
+        "type": "user"
+    }
+    url = "https://api.notion.com/v1/users?page_size=100&start_cursor=test_cursor_2"
+    requests_mock.get(url, json=response_body)
+
+    stream = Users(config=MagicMock())
+
+    user_ids = set()
+    for record in stream.read_records(sync_mode=SyncMode.full_refresh):
```

Comment:
Is there a need to fill the **_user_ids_** set before asserting that the set is of length 220? wouldn't the following snippet do the job? Might also save you some memory from allocating an additional data structure. 
```
records = stream.read_records(sync_mode=SyncMode.full_refresh):
records_length = sum(1 for record in records)
assert records_length == 220
```

Reply:
Updated in [fda250e](https://github.com/airbytehq/airbyte/pull/11452/commits/fda250ecf078624eff22eb745c3eaa7337c947ce).